website/sidebars.json

```diff
@@ -11,7 +11,8 @@
       "runtime",
       "rbi",
       "cli",
-      "tconfiguration"
+      "tconfiguration",
+      "metaprogramming-plugins"
```

Comment:
I'm not a fan of this doc living so close to these other docs. The docs are currently roughly ordered in the sidebar the way that we want people to look at them.

What do you think about adding a new section below like `Advanced` or `Experimental` at the bottom of this list?

I think it would make sense to move the other experimental docs there too, and might be a good place for us to document generics if we wanted to do that. It's fine to just have only DSL plugins be there for now, and I can reorganize things later.